chalice/utils.py

```diff
@@ -7,6 +7,27 @@
 from typing import IO, Dict, Any  # noqa
 
 
+def remove_stage_from_deployed_values(key, filename):
+    # type: (str, str) -> None
+    """Delete a top level key from the deployed JSON file."""
+    final_values = {}  # type: Dict[str, Any]
+    try:
+        with open(filename, 'r') as f:
+            final_values = json.load(f)
+    except IOError:
```

Comment:
I'd prefer we just stick with a single way of dealing with files, either as os.path.isfile() used in `record_deployed_values` or the try/except here.  I'm also pretty sure you can get OSErrors as well.